src/main/java/Parser.java

```diff
@@ -0,0 +1,134 @@
+import java.time.LocalDateTime;
+import java.util.ArrayList;
+
+public class Parser {
+    /**
+     * Parses user input based on different type of command.
+     * For each case, different parse requirement is needed
+     *
+     * @param taskList TaskList of the current file.
+     * @param ui Ui of the project.
+     * @param storage Storage of the project.
+     * @param inputType InputType of the user.
+     * @param userInput The rest of user input after the inputType.
+     * @throws DukeException If user input is not in the format
+     */
+    public static void parse(TaskList taskList, Ui ui, Storage storage, String inputType, String userInput) throws DukeException {
+        if (inputType.equals("todo")) {
+            try {
+                String description = userInput.trim();
+                if (description.isBlank()) {
+                    throw new IllegalArgumentException();
+                }
+                Task newTodo = new Todo(description, false);
+                taskList.addTask(newTodo);
+                ui.printAddTask(taskList, newTodo);
+            } catch (IllegalArgumentException e) {
+                throw new DukeException("OOPS!!! The description of a todo cannot be empty.");
+            } catch (Exception e) {
+                throw new DukeException("OOPS!!! Your input format is wrong. Use: todo [task description]");
+            }
+        } else if (inputType.equals("list")) {
+            try {
+                if (!userInput.isBlank()) {
+                    throw new Exception();
```

Comment:
Perhaps you could directly throw a `DukeException` instead of throwing an `Exception`, catching it, and throwing another `DukeException`? While functionally same, this adds to the verbosity of the code. The use of the superclass `Exception` may also hinder future debugging when new sources of exceptions may arise.